GitTfs.Vs2017/TfsHelper.Vs2017.cs

```diff
@@ -0,0 +1,129 @@
+﻿using System;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices;
+using Microsoft.TeamFoundation.Build.Client;
+using Microsoft.TeamFoundation.Client;
+using Microsoft.TeamFoundation.Server;
+using Microsoft.TeamFoundation.VersionControl.Client;
+using Microsoft.VisualStudio.Setup.Configuration;
+using Sep.Git.Tfs.Core.TfsInterop;
+using Sep.Git.Tfs.VsCommon;
+using StructureMap;
+
+namespace Sep.Git.Tfs.Vs2017
+{
+    public class TfsHelper : TfsHelperBase
+    {
+        private const int REGDB_E_CLASSNOTREG = unchecked((int)0x80040154);
+
+        private const string tfsExtensionsFolder =
+            @"Common7\IDE\CommonExtensions\Microsoft\TeamFoundation\Team Explorer";
+
+        protected string TfsVersionString { get { return "15.0"; } }
+
+        public TfsHelper(TfsApiBridge bridge, IContainer container)
+            : base(bridge, container)
+        { }
+
+        protected override bool HasWorkItems(Changeset changeset)
+        {
+            return Retry.Do(() => changeset.AssociatedWorkItems.Length > 0);
+        }
```

Comment:
Yep, I originally started out inheriting from that class, but much has changed between VS2012-2015 that this base class doesn't make sense to use for 2017. For example, there are method signature changes in version 15 of the Team Foundation client libraries which are breaking changes from version 14.

Also, I'm going to put this PR on hold. While I think checking in will work, my organization uses a comment check-in policy, and I need to find a way to get that in git-tfs; but the current architecture of git-tfs made an assumption (not necessarily a bad one, but unfortunately things changed) that VS' installation folder structure would remain nearly constant--and that's simply not the case with VS2017. And because of the current architecture, I'm unable to change the event handler for assembly resolution because it's buried in the base class constructor.

Having said all of that, I hope at least we can get .Net 4.6.2 on the AppVeyor build server so that when this PR is fully ready, it can be built, tested, and merged.